projects/components/src/action-menu/action-search.provider.spec.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { MockTranslationService } from '@vcd/i18n';
+import { ActionItem } from '../common/interfaces/index';
+import { SpotlightSearchService } from '../spotlight-search/index';
+import { ActionSearchProvider } from './action-search.provider';
+
+interface HasActionSearchProvider {
+    actionSearchProvider: ActionSearchProvider<any, any>;
+}
+
+const MOCK_DATA: ActionItem<any, any>[] = [
+    {
+        textKey: 'Contextual 1',
+        handler: () => console.log('Contextual 1'),
+    },
+    {
+        textKey: 'power.actions',
+        children: [
+            {
+                textKey: 'Start',
+                handler: () => console.log('Starting '),
+            },
+            {
+                textKey: 'Stop',
+                handler: () => console.log('Stopping '),
+            },
+        ],
+    },
+];
+
+describe('ActionSearchProvider', () => {
+    beforeEach(function(this: HasActionSearchProvider): void {
+        this.actionSearchProvider = new ActionSearchProvider(
+            new SpotlightSearchService(),
+            new MockTranslationService()
+        );
+    });
+    describe('actions', () => {
+        it('updates the flattened list of available actions when set', function(this: HasActionSearchProvider): void {
+            expect((this.actionSearchProvider as any).flatListOfAvailableActions).toEqual([]);
+            this.actionSearchProvider.actions = MOCK_DATA;
+            expect((this.actionSearchProvider as any).flatListOfAvailableActions).toEqual(
+                (this.actionSearchProvider as any).getFlatListOfAvailableActions(MOCK_DATA)
+            );
+        });
+    });
+
+    describe('selectedEntities', () => {
+        it('calls the updateFlatListOfAvailableActions method when set', function(this: HasActionSearchProvider): void {
+            const spy = spyOn(this.actionSearchProvider as any, 'updateFlatListOfAvailableActions').and.callFake(
+                () => null
+            );
+            this.actionSearchProvider.selectedEntities = [];
+            expect(spy).toHaveBeenCalled();
+        });
+    });
+
+    describe('actionProviderName', () => {
+        it('calls the register method when set', function(this: HasActionSearchProvider): void {
```

Comment:
It is not important that setting the provider name will call register or not. It is important that this name willl be reflected in the spotlight search service

Reply:
But that is not the job of the `ActionSearchprovider` to make sure about that right. Once it is passed to the SpotlightSearch service it is the job of that to make sure about this